### ISSUE TITLE
Raise friendly errors in case of `num_epochs` is less than `num_warmup_epochs` in CIFAR10 and Transformers examples

### ISSUE THREAD

**Issue by KickItLikeShika** [pytorch/ignite#2207]
I was trying to benchmark `torch.inference_mode` vs `torch.no_grad` on CIFAR10 example, so I just wanted to run for 3 epochs and average the benchmarks, the default warmup epochs is 4, so the scheduler will raise this error, which is not very clear to new users 
```
f"Milestones should be increasing integers, but given {pair[0]} is smaller "
ValueError: Milestones should be increasing integers, but given 291 is smaller than the previous milestone 388
```
So I suggest to check `num_epochs` vs `num_warmup_epochs` inside `run` method and raise more clear error message for the users.

**Comment by KickItLikeShika** [comment]
We have these 2 examples:
- https://github.com/pytorch/ignite/tree/master/examples/contrib/cifar10
- https://github.com/pytorch/ignite/tree/master/examples/contrib/transformers

in both, if the num_epochs is less the num_warmup_epochs, the error in the description will be raised, 
we want to check inside the `rum` method in bot if `num_epochs` is less than `num_warmupds_epochs` we should raise a friendly error message explaining what's the error and why, and what the user should do (either increase the `num_epochs` or decrease `num_warmup_epochs`)

**Comment by KickItLikeShika** [comment]
Hey @himanshu007-creator, thanks a lot for showing interest, we have these 2 examples:
- https://github.com/pytorch/ignite/tree/master/examples/contrib/cifar10
- https://github.com/pytorch/ignite/tree/master/examples/contrib/transformers

in both, if the num_epochs is less the num_warmup_epochs, the error in the description will be raised, 
we want to check inside the `rum` method in bot if `num_epochs` is less than `num_warmupds_epochs` we should raise a friendly error message explaining what's the error and why, and what the user should do (either increase the `num_epochs` or decrease `num_warmup_epochs`)

**Comment by FarehaNousheen** [comment]
Please brief me about the issue a little more. I'll work on this issue.

**Comment by KickItLikeShika** [comment]
Hi @FarehaNousheen, please check the comment above https://github.com/pytorch/ignite/issues/2207#issuecomment-923125971, it explains what we need to do in depth. If you still can't understand what we need to do, please ping me on discord.

**Comment by DebarshiChanda** [comment]
Hey @KickItLikeShika, I think adding an assertion can solve this issue. Can you assign this to me?

**Comment by theory-in-progress** [comment]
Hey @KickItLikeShika, I think I've made the necessary changes in three of the files to raise an user friendly message when ``num_epochs`` is greater than ``num_warmup_epochs``. Am I free to create a Pull Request, since this hasn't been assigned to me...

**Comment by vfdev-5** [comment]
@DebarshiChanda @theory-in-progress to be fair if someone is assigned to the issue his/her PR has higher prio (conditioned also by the time it takes to work on the issue). Thanks a lot to you both on working on this tiny issue ! In this particular case, @theory-in-progress 's PR looks better and we'll proceed with it.
@DebarshiChanda please pick another issue to work on, thanks a lot !

**Comment by theory-in-progress** [comment]
Of course! sorry for barging in... Thanks for the opportunity @vfdev-5! I'm happy to work on this issue
